sql/update.go

```diff
@@ -44,7 +45,31 @@ func (p *planner) Update(n *parser.Update) (planNode, error) {
 	// Determine which columns we're inserting into.
 	var names parser.QualifiedNames
 	for _, expr := range n.Exprs {
-		names = append(names, expr.Name)
+		var err error
+		expr.Expr, err = p.expandSubqueries(expr.Expr, len(expr.Names))
+		if err != nil {
+			return nil, err
+		}
+
+		if expr.Tuple {
+			// TODO(pmattis): The distinction between Tuple and DTuple here is
+			// irritating. We'll see a DTuple if the expression was a subquery that
+			// has been evaluated. We'll see a Tuple in other cases.
+			n := 0
+			switch t := expr.Expr.(type) {
+			case parser.Tuple:
+				n = len(t)
+			case parser.DTuple:
+				n = len(t)
```

Comment:
I'm pretty sure you can write this as one case parser.Tuple, parser.DTuple even when they are different types.